include/wil/registry_basic.h

```diff
@@ -0,0 +1,1540 @@
+{
+    const reg_view_details::reg_view regview{key};
+    return regview.create_shared_key(path, access, security_descriptor);
+}
+#endif
+#endif
+
+inline HRESULT open_key_nothrow(HKEY key, _In_opt_ PCWSTR path, ::wil::unique_hkey& hkey, ::wil::registry::key_access access = ::wil::registry::key_access::read) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.open_key(path, hkey, access);
+}
+
+#if defined(__WIL_WINREG_STL)
+inline HRESULT open_key_nothrow(HKEY key, _In_opt_ PCWSTR path, ::wil::shared_hkey& hkey, ::wil::registry::key_access access = ::wil::registry::key_access::read) WI_NOEXCEPT
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.open_key(path, hkey, access);
+}
+#endif
+
+inline HRESULT create_key_nothrow(HKEY key, _In_ PCWSTR path, ::wil::unique_hkey& hkey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr)
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.create_key(path, hkey, access, security_descriptor);
+}
+
+#if defined(__WIL_WINREG_STL)
+inline HRESULT create_key_nothrow(HKEY key, _In_ PCWSTR path, ::wil::shared_hkey& hkey, ::wil::registry::key_access access = ::wil::registry::key_access::read, _In_opt_ PCWSTR security_descriptor = nullptr)
+{
+    const reg_view_details::reg_view_nothrow regview{key};
+    return regview.create_key(path, hkey, access, security_descriptor);
+}
+#endif
+
+#ifdef WIL_ENABLE_EXCEPTIONS
+inline size_t get_child_key_count(HKEY key)
+{
+    DWORD numSubKeys{};
+    THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+        key, nullptr, nullptr, nullptr, &numSubKeys, nullptr,
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr));
+    return numSubKeys;
+}
+
+inline size_t reg_get_child_value_count(HKEY key)
+{
+    DWORD numSubValues{};
+    THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+        key, nullptr, nullptr, nullptr, nullptr, nullptr,
+        nullptr, &numSubValues, nullptr, nullptr, nullptr, nullptr));
+    return numSubValues;
+}
+#endif
+
+inline HRESULT get_child_key_count_nothrow(HKEY key, _Out_ DWORD* numSubKeys) WI_NOEXCEPT
+{
+    RETURN_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+        key, nullptr, nullptr, nullptr, numSubKeys, nullptr,
+        nullptr, nullptr, nullptr, nullptr, nullptr, nullptr));
+    return S_OK;
+}
+
+inline HRESULT reg_get_child_value_count_nothrow(HKEY key, _Out_ DWORD* numSubValues) WI_NOEXCEPT
+{
+    RETURN_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+        key, nullptr, nullptr, nullptr, nullptr, nullptr,
+        nullptr, numSubValues, nullptr, nullptr, nullptr, nullptr));
+    return S_OK;
+}
+
+// Registry Set* functions, both throwing and non-throwing
+
+#ifdef WIL_ENABLE_EXCEPTIONS
+inline void set_value_dword(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD data)
+{
+    const reg_view_details::reg_view regview{key};

```

Comment:
We discussed this, but trying to capture it here: 

We should see if we can invert this calling structure---IMO, the `reg_view` should call the free functions to get the data. That way, folks calling the free functions directly do not need to pay the cost of instantiating a reg_view.